src/components/Form/Widgets/PasswordWidget.tsx

```diff
@@ -25,19 +24,28 @@ export interface PasswordStrengthProps {
 }
 
 const PasswordStrength = ({ password }: PasswordStrengthProps) => {
-	const [strengthScore, setStrengthScore] = React.useState<
-		number | undefined
-	>();
+	const [strengthScore, setStrengthScore] = useState<number | undefined>();
 
-	React.useEffect(() => {
-		if (password) {
+	useEffect(() => {
+		// Since the dynamic import is asynchronous we need to set a variable. With the dynamic import we might end up
+		// updating the state of an unmounted component which would lead to memory leaks, as it can prevent React from properly cleaning up the component.
+		let isMounted = true;
+
+		(async () => {
 			try {
-				const { score } = zxcvbn(password);
-				setStrengthScore(score);
-			} catch {
-				// Ignore any errors, as we only want to show the strength meter if it is available.
+				const zxcvbn = (await import('zxcvbn')).default;
```

Comment:
We probably only have to import it only the first time that the password isn't empty right? (I realize it shouldn't make much difference, but in any case it would be 1 less function call).

Reply:
How about using 1 useEffect (w/ a useRef?) to load `zxcvbn` the first time the password has a value and this second useEffect just for the calculation?